### PR TITLE
Modification of applyAuction for new account model #94

### PR DESCRIPTION
**Pull request by josojo**
Tackles #94 

Idea:
Instead of checking the nonces,
`require(
    sha(auctions[slot].shaHash, nonce_account_1, …, nonce_account_50) == _orderHash 
);`
 I am checking the hashes, as we no longer have a unique mapping between nonce==currentBatchIndex

test plan:
unit-tests only

### CODE REVIEW

File: contracts/SnappAuction.sol
Comment (fleupold) [inline contracts/SnappAuction.sol]: This would work. Unfortunately it would cost 100k gas to send these 50 indices.

File: contracts/SnappAuction.sol
Comment (bh2smith) [inline contracts/SnappAuction.sol]: Should this be `encodePacked`? Why or why not?

File: contracts/SnappAuction.sol
Comment (bh2smith) [inline contracts/SnappAuction.sol]:
```suggestion
            );
```

File: contracts/SnappAuction.sol
Comment (bh2smith) [inline contracts/SnappAuction.sol]:
```suggestion
        return sha256(abi.encodePacked(batchHashSequence));
```

File: contracts/SnappAuction.sol
Comment (bh2smith) [inline contracts/SnappAuction.sol]: I think this entire function can be replaced with 

```js
return _auctionIndex >= getStandingOrderValidFrom(userId, orderBatchIndex) && _auctionIndex <= getStandingOrderValidTo(userId, orderBatchIndex);
```

File: contracts/SnappAuction.sol
Comment (josojo) [reply contracts/SnappAuction.sol]: It will not make a difference... on bytes32, they are operating the same way

Comment (fleupold) [reply contracts/SnappAuction.sol]: You are right. I'd still prefer to stick to use the same in all of our codebase.

Comment (fleupold) [reply contracts/SnappAuction.sol]: nit: maybe it's simpler if we extract all orderHashes into an array of bytes32 and only pack them once?

**Review by fleupold — COMMENTED** [review]
Looks good to me overall. Nice unit tests!

Would it make sense to start building driver support for this before we land so that we can quickly get the services depend on master as soon as we merge this?

**Comment by josojo** [comment]
Yes, I would love to do the driver support for this. Please! But you can do it as well, don't know how much time I will spend at the conference in the next two days.

**Comment by josojo** [comment]
ToDo: The function `calculateOrderHash(uint slot, uint128[] memory _standingOrderIndex)` should also take the current `orderhash` of the regular accounts into account

done

**Comment by josojo** [comment]
@fleupold @bh2smith Can I get approval for this PR?

I will open a PR for changing the code to work with u32 ipo U128

**Review by fleupold — APPROVED** [review]
Looks good to me

File: contracts/SnappAuction.sol
Comment (bh2smith) [inline contracts/SnappAuction.sol]:
```suggestion
        );
```

**Review by bh2smith — APPROVED** [review]
Very nice! Just one indentation correction.

**Pull request merged**
